patchlevel.h

```diff
@@ -39,9 +39,32 @@ Instead use one of the version comparison macros.  See C<L</PERL_VERSION_EQ>>.
 /* do not adjust the whitespace! Configure expects the numbers to be
  * exactly on the third column */
 
-#define PERL_REVISION	5		/* age */
-#define PERL_VERSION	33		/* epoch */
-#define PERL_SUBVERSION	1		/* generation */
+/* Perl semantic versioning
+   you should avoid using these variables directly
+   and prefer using the compare functions.
+       - PERL_VERSION_EQ
+       - PERL_VERSION_NE
+       - PERL_VERSION_LT
+       - PERL_VERSION_GT
+       - PERL_VERSION_LE
+       - PERL_VERSION_GE
+*/
+#define PERL_VERSION_MAJOR	      7
+#define PERL_VERSION_MINOR	      1
+#define PERL_VERSION_PATCH	      0
+#define PERL_VERSION_RELEASE      0 /* for future usage */
+#define PERL_VERSION_BUILD        0 /* for future usage */
+
+/* Legacy Perl5 VERSION macro hardcoded to 5.255.255
+   You should never change these values
+   This fixes two XS Perl5 modules issues:
+    - modules using an old version of ppport.h
+    - incomplete version checks when only using PERL_VERSION for example
+   view define GCC_DIAG_PRAGMA(x) _Pragma (#x)
```

Comment:
What's this line doing here? It seems like a stray compiler directive that's got lost into a comment

Reply:
nice catch, fixed by 87bd1e47ee5de335ffdb309fc8cacf5f7e7b584a 
will amend later